tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -15840,8 +15840,10 @@ void SpirvEmitter::addDerivativeGroupExecutionMode() {
   assert(spvContext.isCS());
 
   SpirvExecutionMode *numThreadsEm =
-      cast<SpirvExecutionMode>(spvBuilder.getModule()->findExecutionMode(
+      dyn_cast<SpirvExecutionMode>(spvBuilder.getModule()->findExecutionMode(
           entryFunction, spv::ExecutionMode::LocalSize));
+  if (!numThreadsEm)
```

Comment:
This if confused me at first. It seemed backwards. Maybe a comment explaining it. Something to the effect that if the local size is not found it must have a local size id, which is handled in differently.